numpy/core/numeric.py

```diff
@@ -289,16 +288,14 @@ def full(shape, fill_value, dtype=None, order='C'):
     >>> np.full((2, 2), np.inf)
     array([[ inf,  inf],
            [ inf,  inf]])
-    >>> np.full((2, 2), 10, dtype=np.int)
+    >>> np.full((2, 2), 10)
     array([[10, 10],
            [10, 10]])
 
     """
+    if dtype is None:
+        dtype = array(fill_value).dtype
```

Comment:
Maybe there should be an error. `arr.fill` kind of tries this, not sure about all the logic it uses.

In any case, this is an orthogonal issue, so I think we should ignore it here.